rust/Cargo.toml

```diff
@@ -31,3 +31,7 @@ rand = "0.8.4"
 [[bench]]
 harness = false
 name = "benches_main"
+
+[lints.rust]
+# Required for Rust >= 1.84 && pyo3 < 0.23
+unexpected_cfgs = { level = "allow", check-cfg = ['cfg(addr_of)'] }
```

Comment:
I would open a ticket about this so that we can keep track of it and actually work towards upgrading to 0.23

Reply:
Renovate will remind us as well :)